src/licensedcode/plugin_license.py

```diff
@@ -116,3 +117,57 @@ def get_scanner(
             license_text_diagnostics=license_text_diagnostics,
             license_url_template=license_url_template
         )
+
+    def process_codebase(self, codebase, **kwargs):
+        
+        if codebase.has_single_resource:
+            return
+
+        for resource in codebase.walk(topdown=False):
+            match_reference_license(resource,codebase)
+
+
+def match_reference_license(resource, codebase):
+    """
+    Return the updated license matches having reference to another files
+    """
+    if not resource.is_file:
+        return
+
+    licenses = resource.licenses
+    license_expressions = resource.license_expressions
+    if not licenses:
+        return 
+
+    referenced_licenses = []
+    referenced_license_expressions = []
+    referenced_filenames = []
+    
+    for license in licenses:
+        referenced_files = license['matched_rule']['referenced_filenames']
+        for referenced_filename in referenced_files:
+            if not referenced_filename in referenced_filenames:
+                referenced_filenames.append(referenced_filename)
+                
+    for referenced_filename in referenced_filenames:
+        new_resource = find_reference_licenses(referenced_filename=referenced_filename, resource=resource, codebase=codebase)
+        if new_resource:
+            referenced_licenses.extend(new_resource.licenses)
+            referenced_license_expressions.extend(new_resource.license_expressions)
+
+    licenses.extend(referenced_licenses)
+    license_expressions.extend(referenced_license_expressions)
+    codebase.save_resource(resource)
+    return resource
+
+
+def find_reference_licenses(referenced_filename, resource, codebase, **kwargs):
+    """
+    Return the resource object for matching referenced-filename given a resource in codebase
```

Comment:
What about this instead?
```suggestion
    Return a Resource matching the ``referenced_filename`` path or filename given a ``resource`` in ``codebase``.
    Return None if the ``referenced_filename`` cannot be found in the same directory as the base ``resource``.
    ``referenced_filename`` is the path or filename referenced in a LicenseMatch of ``resource``,
```